config-connector/solutions/iam/helm/service-account/templates/iampolicymember.yaml

```diff
@@ -0,0 +1,25 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMPolicyMember
+metadata:
+  name: iampolicymember-service-account
+spec:
+  member: {{ required "IAM member is required!" .Values.iamPolicyMember.iamMember }}
+  role: {{ .Values.iamPolicyMember.role | default "roles/iam.serviceAccountKeyAdmin" }}
+  resourceRef:
+    apiVersion: iam.cnrm.cloud.google.com/v1beta1
+    kind: IAMServiceAccount
+    name: {{ .Values.serviceAccount.name | default "service-account-solution" }}
```

Comment:
I would view trying to make it exactly same as optional, as long as we follow the structure and extent of customization: same configuration is produced, only setters and same setters.

Reply:
Thanks for the thoughts. I removed the default values, so there is no pipe feature used as well.

I just think that if we apply some helm specific feature in the solution, it would not hurt the user experience. Kpt and helm are two different things, I think some particularity are acceptable to me as a user. But it's minor since we are keeping the solution simple and the feature would not be applied.

Reply:
@AlexBulankou's comment makes sense.
@MingyanZhao Yes, agreed that keeping consistent user experience (consistent with other Helm charts) is good. Feel free to apply Helm's good practices in the solutions as long as the setters are the same.

Reply:
Here I removed the default values. By doing this means the user needs to check out the values.yaml to find out what are the default values. So that the values.yaml could be the values table for the solution where all the required fields and optional fields are shown in the same file.